app/models/govuk_publishing_components/component_docs.rb

```diff
@@ -19,6 +19,10 @@ def used_in_this_app
       fetch_component_docs.map { |component| build(component) if component_in_use(component[:id]) }.compact.sort_by(&:name)
     end
 
+    def not_used_in_this_app
+      fetch_component_docs.map { |component| build(component) if component_not_in_use(component[:id]) }.compact.sort_by(&:name)
+    end
+
```

Comment:
I wonder if we could reduce a bit of duplication here?

maybe something like this:
```
    def used_in_this_app
      fetch_component_docs.map { |component| build(component) if component_in_use(component[:id]) }.compact.sort_by(&:name)
    end

    def not_used_in_this_app
      fetch_component_docs.map { |component| build(component) if component_not_in_use(component[:id]) }.compact.sort_by(&:name)
    end
    
    private
        def filter_by_usage(in_use: true)
          fetch_component_docs.map { |component|
        build(component) if component_in_use?(component[:id], in_use:)
      }.compact.sort_by(&:name)
      end
```

What do you think? is it over complicated?